TypeError: _validate_attrs() missing 1 required positional argument: 'engine'

### What happened?

An upstream change to xarray's `validate_attrs` function has broken `canonical_dataset_to_grib`. This change has not been released yet but will in the coming days. 

See https://github.com/pydata/xarray/pull/9407 for more details.

### What are the steps to reproduce the bug?

The traceback is below:

```
    def canonical_dataset_to_grib(dataset, path, mode="wb", no_warn=False, grib_keys={}, **kwargs):
        # type: (xr.Dataset, str, str, bool, T.Dict[str, T.Any], T.Any) -> None
        """
        Write a ``xr.Dataset`` in *canonical* form to a GRIB file.
        """
        if not no_warn:
            warnings.warn("GRIB write support is experimental, DO NOT RELY ON IT!", FutureWarning)
    
        # validate Dataset keys, DataArray names, and attr keys/values
        xr.backends.api._validate_dataset_names(dataset)  # type: ignore
>       xr.backends.api._validate_attrs(dataset)  # type: ignore
E       TypeError: _validate_attrs() missing 1 required positional argument: 'engine'
```



### Version

0.9.14.0

### Platform (OS and architecture)

linux

### Relevant log output

_No response_

### Accompanying data

_No response_

### Organisation

_No response_

--- a/cfgrib/xarray_to_grib.py
+++ b/cfgrib/xarray_to_grib.py
@@ -253,7 +253,11 @@ def canonical_dataset_to_grib(dataset, path, mode="wb", no_warn=False, grib_keys
 
     # validate Dataset keys, DataArray names, and attr keys/values
     xr.backends.api._validate_dataset_names(dataset)  # type: ignore
-    xr.backends.api._validate_attrs(dataset)  # type: ignore
+    # _validate_attrs takes the engine name as its 2nd arg from xarray 2024.09.0
+    try:
+        xr.backends.api._validate_attrs(dataset)  # type: ignore
+    except TypeError:
+        xr.backends.api._validate_attrs(dataset, "cfgrib")  # type: ignore
 
     real_grib_keys = {str(k)[5:]: v for k, v in dataset.attrs.items() if str(k)[:5] == "GRIB_"}
     real_grib_keys.update(grib_keys)

